nixos/munin: fix two issues with extraAutoPlugins

ToxicFrog opened the pull request:
## Description of changes

If extraAutoPlugins contains values that carry context (e.g. it comes from a flake input), the keys generated from them using baseNameOf inherit that context and the config doesn't compile.

This doesn't actually need to be an attrset anyways, so a bit of internal refactoring lets us fix this without changing the visible API.

It also tweaks the `sed` command in `internAndFixPlugins` to not mangle nix store paths, so it doesn't break plugins that are coming from nix tools like `writeShellScriptBin` rather than from munin-contrib.

Relevant: #17895 -- this makes it easier to use munin-contrib plugins from flake inputs or from the store. (Contrib still isn't packaged though.)

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

ToxicFrog committed:
nixos/munin: fix key-context error when using extraAutoPlugins

If extraAutoPlugins contains values that carry context (e.g. it comes
from a flake input), the keys generated from them using baseNameOf
inherit that context and the config doesn't compile.

This doesn't actually need to be an attrset anyways, so a bit of
internal refactoring lets us fix this without changing the visible API.

bjornfor commented on nixos/modules/services/monitoring/munin.nix:
```diff
@@ -83,14 +83,15 @@ let
-  internManyPlugins = name: path:
+  # store, with no renaming. The output is suitable for use with
+  # munin-node-configure --suggest, i.e. munin.extraAutoPlugins.
+  # Note that this flattens the input; this is intentional, as node-configure
```
node-configure -> munin-node-configure?

ToxicFrog replied:
Whoops, done. Thanks.

bjornfor commented on nixos/modules/services/monitoring/munin.nix:
```diff
@@ -375,7 +372,7 @@ in
     # Munin is hardcoded to use DejaVu Mono and the graphs come out wrong if
     # it's not available.
-    fonts.packages = [ pkgs.dejavu_fonts ];
+    fonts.fonts = [ pkgs.dejavu_fonts ];  # fonts.packages in unstable, but fonts.fonts in 2305
```
Unrelated change. Please put in a separate commit. (Same PR is fine.)

ToxicFrog replied:
This is what I get for using `git add` without `-p` late at night -- that change shouldn't be merged, it's a local change to let my (nixos-23.03) systems load this (nixos-unstable) module.

I'll revert that, and I also found another munin-module bug related to `internAndFixPlugins` that I want to get a fix for into this branch later this afternoon.

ToxicFrog committed:
nixos/munin: fix internAndFixPlugins mangling store paths

The previous implementation works fine when the plugins do not already
contain store paths, which is the case for stuff from munin-contrib.
However, for plugins generated via nix (e.g. with writeShellScriptBin),
it tries to fix the paths in it which already point to the nix store,
ruining everything.

bjornfor commented on nixos/modules/services/monitoring/munin.nix:
```diff
@@ -96,14 +96,22 @@ let
-      find . -type f -exec sed -E -i '
-        s,(/usr)?/s?bin/,/run/current-system/sw/bin/,g
+      ${pkgs.findutils}/bin/find . -type f -exec ${pkgs.gnused}/bin/sed -E -i '
+        \%/nix/store/%! s,(/usr)?/s?bin/,/run/current-system/sw/bin/,g
```
Use $NIX_STORE (?) env var instead of hardcoding the path?

ToxicFrog replied:
This sent me down a bit of a rabbit hole, because it's `$NIX_STORE_PATH` when *invoking* nix (and not guaranteed to be set), but `$NIX_PATH` once the builder actually runs (and is guaranteed). Fixed, thanks.

bjornfor commented on nixos/modules/services/monitoring/munin.nix:
```diff
@@ -96,15 +96,23 @@ let
   # Use the appropriate intern-fn to copy the plugins into the store and patch
   # them afterwards in an attempt to get them to run on NixOS.
+  # This is a bit hairy because we can't just fix shebangs; lots of munin plugins
+  # hardcode paths like /sbin/mount rather than truing $PATH, so we have to look
```
truing -> trusting?

ToxicFrog replied:
Done.

bjornfor commented:
Thanks!
